protocol/messenger_handler.go

```diff
@@ -1622,11 +1622,20 @@ func (m *Messenger) HandleCommunityRequestToJoinResponse(state *ReceivedMessageS
 			communitySettings := response.CommunitiesSettings()[0]
 			community := response.Communities()[0]
 			state.Response.AddCommunity(community)
+			chatIDs, err := m.getChatIdsForCommunity(community.ID())
+			if err != nil {
+				return err
+			}
+
+			if len(chatIDs) > 0 {
+				generalChannelChatID := m.Chat(chatIDs[0])
+				state.Response.AddChat(generalChannelChatID)
+			}
+
```

Comment:
I'm not sure what you mean @osmaczko : would you mind elaborating?

Reply:
Sure. We can [spectate](https://github.com/status-im/status-go/blob/058120fad95056627badd8854b6655d70258db4c/protocol/messenger_communities.go#L736) community before we join it. For open communities we should see general channel in that case as well.

Reply:
Please correct me If I'm wrong, but I think that already happens because of this line here : https://github.com/status-im/status-go/blob/058120fad95056627badd8854b6655d70258db4c/protocol/messenger_communities.go#L751 

On the mobile client we do see the general channel when we request to join a community (after it has been shared).

The updated approach now in this PR is to merge responses (messenger response + response to joining community) so that chat information (i.e channel information) was not missed in that particular response signal. 
In the case of open community it is the general channel.